modules/mini-games/Race.lua

```diff
@@ -2,86 +2,69 @@ local Mini_games        = require "expcore.Mini_games"
 local player_progress = {}
 local cars = {}
 local variables = {}
-local race = Mini_games.new_game("Race_game")
-local token_for_car
 local scores = {}
 local laps = {}
 local gate_boxes = {}
 local start_players ={}
 
-local function setup_areas()
-    if not areas[1] then
-        areas[1] = surface[1].get_script_areas("gate_1_box")[1].area
-        areas[2] = surface[1].get_script_areas("gate_2_box")[1].area
-        areas[3] = surface[1].get_script_areas("gate_3_box")[1].area
-        areas[4] = surface[1].get_script_areas("gate_4_box")[1].area
-    end
-    if not gate_boxes[1] then
-        gate_boxes[1] = surface[1].get_script_areas("gate_1")[1].area
-        gate_boxes[2] = surface[1].get_script_areas("gate_2")[1].area
-        gate_boxes[3] = surface[1].get_script_areas("gate_3")[1].area
-        gate_boxes[4] = surface[1].get_script_areas("gate_4")[1].area
-    end
-
-    variables["finsih"] = surface[1].get_script_areas("finsish_line")[1].area
-
-    --gate
-    if not gates[1] then
-        gates[1] = surface[1].find_entities_filtered {area = gate_boxes[1], name = "gate"}
-        gates[2] = surface[1].find_entities_filtered {area = gate_boxes[2], name = "gate"}
-        gates[3] = surface[1].find_entities_filtered {area = gate_boxes[3], name = "gate"}
-        gates[4] = surface[1].find_entities_filtered {area = gate_boxes[4], name = "gate"}
-    end
-end
-
-
+--- Register a new permission group for players not in cars
 Permission_Groups.new_group('out_car')
 :disallow_all()
 :allow{
     'write_to_console'
 }
 
+--- Register a new debug interface to store all local variables
+local debug_interface = {}
+interface.add_interface_module('Race', debug_interface)
 
+--- Register all global variables
 Global.register({
-    surface = surface,
-    gates = gates,
-    variables = variables,
-    areas = areas,
+    surface         = surface,
+    gates           = gates,
+    variables       = variables,
+    areas           = areas,
     player_progress = player_progress,
-    cars = cars,
-    scores = scores,
-    laps = laps,
-    gate_boxes = gate_boxes,
-    start_players = start_players
+    cars            = cars,
+    scores          = scores,
+    laps            = laps,
+    gate_boxes      = gate_boxes,
+    start_players   = start_players
 },function(tbl)
-    surface = tbl.surface
-    gates = tbl.gates
-    variables = tbl.variables
-    areas = tbl.areas
+    surface         = tbl.surface
+    gates           = tbl.gates
+    variables       = tbl.variables
+    areas           = tbl.areas
     player_progress = tbl.player_progress
-    cars = tbl.cars
-    scores = tbl.scores
-    laps = tbl.laps
-    gate_boxes = tbl.gate_boxes
-    start_players = tbl.start_players
+    cars            = tbl.cars
+    scores          = tbl.scores
+    laps            = tbl.laps
+    gate_boxes      = tbl.gate_boxes
+    start_players   = tbl.start_players
+    for k, v in pairs(tbl) do
+        debug_interface[k] = v
+    end
```

Comment:
i do quite like this 👍

Reply:
Thanks, its much better than writing out the whole table by hand.